src/main/java/org/skriptlang/skript/lang/properties/PropertyRegistry.java

```diff
@@ -0,0 +1,70 @@
+package org.skriptlang.skript.lang.properties;
+
+import ch.njol.skript.Skript;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Unmodifiable;
+import org.skriptlang.skript.util.Registry;
+
+import java.util.*;
+
+/**
+ * A registry handling {@link Property}s.
+ * Skript and addons should register any new properties here to avoid conflicts.
+ * If a conflict does occur, your property will fail to register and return false. You should check if your property
+ * has a matching handler with the already registered one. If so, you should be able to use the other property instead of your own.
+ */
+public class PropertyRegistry implements Registry<Property<?>> {
```

Comment:
Might be good to implement ViewProvider for this. Would require changes to the existing registration process though (e.g you'll need to register the default properties through an AddonModule)